base/fhir_parser.bal

```diff
@@ -166,8 +167,11 @@ isolated function parseFHIRResource(Profile profile, json|xml payload) returns a
     if payload is json {
         anydata|error cloneWithType = payload.cloneWithType(profile.modelType);
         if cloneWithType is error {
+            map<value:Cloneable> & readonly detail = cloneWithType.detail();
+            string|error detailedMessage = detail.get("message").ensureType(string);
+            string diagnostic = detailedMessage is string ? detailedMessage : cloneWithType.detail().toString();
```

Comment:
The value of `cloneWithType.detail()` is stored in the `details` variable, so we can re-use it in other places?

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/294e8d4a7ce4b2a246db0c141b9db13d6a2f05ad